functions/index.js

```diff
@@ -102,7 +104,7 @@ app.get('/dpr-aware-image', (req, res) => {
     request.get(url).pipe(res);
   } catch (error) {
     console.log('[server dpr-aware-image request proxy] error => ', error);
-    res.json({error});
+    res.status(500).end(SERVER_ERROR_MESSAGE);
```

Comment:
Could you share why a custom error handler would be better here than `res.render('error', { error: err })`? Just wondering if we should supply more information than `Something went wrong!` :)

Reply:
Actually for this kind of implementation, I'd like to listen to your professional opinion.

From my experience, I thought that we should not expose backend information to the frontend for the reason of security or something. So I just used to define some error template with the proper status code so that only debugging on backend can give the answer to the error.
To be honest, it's my own opinion. So I do like to listen to you and learn from you.

Reply:
When a demo or service is mostly user-facing (e.g someone that is not a developer), I attempt to provide as user-friendly an error message as possible with some idea of why something went wrong and what their next step should be. Given this is a developer-focused demo, I think providing as much debug assistance as possible would make more sense.

I would personally go for `res.render('error', { error: err })` :)

Reply:
Yes, I'll follow you.